### PR TITLE
use task level export destinations

### PR DESCRIPTION
**Pull request by jackschofield23**
### Description

changes export destinations to use the task level export, and validates against the 
Fixes # .

A few sentences describing the changes proposed in this pull request.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.

### CODE REVIEW

File: src/Contracts/Models/TaskObject.cs [hunk 1]
```diff
@@ -39,7 +39,7 @@ public class TaskObject
         public TaskDestination[] TaskDestinations { get; set; } = System.Array.Empty<TaskDestination>();
 
         [JsonProperty(PropertyName = "export_destinations")]
-        public TaskDestination[] ExportDestinations { get; set; } = System.Array.Empty<TaskDestination>();
+        public ExportDestination[] ExportDestinations { get; set; } = System.Array.Empty<ExportDestination>();
```
Comment (lillie-dae) [inline src/Contracts/Models/TaskObject.cs:42]: should we add using System; in this file to shorten these System.Arrays... ?

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 2]
```diff
@@ -319,25 +318,40 @@ private async Task<bool> UpdateWorkflowInstanceStatus(WorkflowInstance workflowI
 
         private async Task HandleDicomExport(WorkflowRevision workflow, WorkflowInstance workflowInstance, TaskExecution task, string correlationId)
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:319]: function name is missing Async
should be HandleDicomExportAsync

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs [hunk 3]
```diff
@@ -319,25 +318,40 @@ private async Task<bool> UpdateWorkflowInstanceStatus(WorkflowInstance workflowI
...
 
-        private string[] GetDicomExports(WorkflowRevision workflow, WorkflowInstance workflowInstance, TaskExecution task)
+        private string[] GetDicomExports(WorkflowRevision workflow, WorkflowInstance workflowInstance, TaskExecution task, string[] exportDestinations)
         {
-            var exportDestinations = workflow.Workflow?.InformaticsGateway?.ExportDestinations;
+            var validExportDestinations = workflow.Workflow?.InformaticsGateway?.ExportDestinations;
 
-            if (exportDestinations is null || !exportDestinations.Any())
+            if (validExportDestinations is null || !validExportDestinations.Any()
+                || exportDestinations is null || !exportDestinations.Any())
```
Comment (lillie-dae) [inline src/WorkflowExecuter/Services/WorkflowExecuterService.cs:342]: I think we have an ICollection extension for this now to do a .IsNullOrEmpty() on arrays and other collections which can simplify this.

**Review by lillie-dae — APPROVED** [review]

File: src/Contracts/Models/TaskObject.cs (hunk 1 above)
Comment (jackschofield23) [reply src/Contracts/Models/TaskObject.cs]: Removed

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs (hunk 3 above)
Comment (jackschofield23) [reply src/WorkflowExecuter/Services/WorkflowExecuterService.cs]: Arrays does not include this, but added an extension method

File: src/WorkflowExecuter/Services/WorkflowExecuterService.cs (hunk 2 above)
Comment (jackschofield23) [reply src/WorkflowExecuter/Services/WorkflowExecuterService.cs]: Changed

**Pull request merged**